### PR TITLE
feat: 1.0 - Use @sentry/browser

### PR DESCRIPTION
**Pull request by HazAT**
This is the tracking PR for unifying the react-native SDK. 

- [x] Use `@sentry/browser`
- [x] Fetch release from native and set in JS
- [x] Fetch device info from native and set in JS
- [x] Use native transport
- [x] Unified API

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # Changelog
 
+## 1.0.0-beta
+
+This is a new major release of the Sentry's React Native SDK rewritten in TypeScript.
+This SDK is now is unified and published under a new name `@sentry/react-native`.
```
Comment (kamilogorek) [inline CHANGELOG.md]:
```suggestion
This SDK is now unified with the rest of our JavaScript SDKs and published under a new name `@sentry/react-native`.
```

File: SentryReactNative.podspec [hunk 2]
```diff
@@ -18,8 +18,8 @@ Pod::Spec.new do |s|
   s.preserve_paths = '*.js'
 
   s.dependency 'React'
-  s.dependency 'Sentry', '~> 4.1.3'
+  s.dependency 'Sentry', '~> 4.3.4'
```
Comment (kamilogorek) [inline SentryReactNative.podspec]: `4.3.4`?

File: android/src/main/java/io/sentry/RNSentryModule.java [hunk 3]
```diff
@@ -159,66 +115,45 @@ public void setLogLevel(int level) {
...
-    public void addExtra(String key, String value) {
-        RNSentryModule.extra.put(key, value);
-        logger.info(String.format("addExtra '%s' '%s'", key, value));
-    }
-
-    @ReactMethod
-    public void setTags(ReadableMap tags) {
-        RNSentryModule.tags = tags;
+    public void crash() {
+        throw new RuntimeException("TEST - Sentry Client Crash");
```
Comment (kamilogorek) [inline android/src/main/java/io/sentry/RNSentryModule.java]: Should this be here?

File: src/js/backend.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  /**
+   * @inheritDoc
+   */
+  protected _setupTransport(): Transport {
+    if (!this._options.dsn) {
+      // We return the noop transport here in case there is no Dsn.
+      return new NoopTransport();
+    }
+
+    const transportOptions = this._options.transportOptions
```
Comment (kamilogorek) [inline src/js/backend.ts]: `{ ...this._options.transportOptions, dsn: this._options.dsn }`

File: src/js/client.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  /**
+   * @inheritDoc
+   */
+  protected _prepareEvent(
+    event: Event,
+    scope?: Scope,
+    hint?: EventHint
+  ): SyncPromise<Event | null> {
+    event.platform = event.platform || "javascript";
```
Comment (kamilogorek) [inline src/js/client.ts]: should `platform` ever be overwriteable?

File: src/js/index.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  Response,
+  Severity,
+  StackFrame,
+  Stacktrace,
+  Status,
+  Thread,
+  User
+} from "@sentry/types";
+
+export {
```
Comment (kamilogorek) [inline src/js/index.ts]: Related: https://github.com/getsentry/sentry-javascript/issues/2101

File: src/js/integrations/devicecontext.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  public setupOnce(): void {
+    addGlobalEventProcessor(async (event: Event) => {
+      const self = getCurrentHub().getIntegration(DeviceContext);
+      if (!self) {
+        return event;
+      }
+
+      try {
+        const deviceContexts = await RNSentry.deviceContexts();
+        event.contexts = { ...deviceContexts, ...event.contexts };
```
Comment (kamilogorek) [inline src/js/integrations/devicecontext.ts]: `devideContext` should probably have priority here, so I'd reverse spread operations

File: src/js/integrations/reactnativeerrorhandlers.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+            scope.setLevel(Severity.Fatal);
+          }
+          getCurrentHub().captureException(error, {
+            originalException: error
+          });
+        });
+
+        const client = getCurrentHub().getClient();
+        if (client) {
+          client.flush(2000).then(() => {
```
Comment (kamilogorek) [inline src/js/integrations/reactnativeerrorhandlers.ts]: It should be controlled by `shutdownTimeout` client option

File: src/js/sdk.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    enableNativeCrashHandling: true
+  }
+): void {
+  if (options.defaultIntegrations === undefined) {
+    options.defaultIntegrations = [
+      new ReactNativeErrorHandlers(),
+      new Release(),
+      ...defaultIntegrations.filter(
+        integration =>
+          integration.name !== "GlobalHandlers" && // We will use the react-native internal handlers
```
Comment (kamilogorek) [inline src/js/sdk.ts]: I'd change that slightly to say

```js
...defaultIntegrations.filter(i => !IGNORED_DEFAULT_INTEGRATIONS.includes(i))
```

and somewhere at the top

```js
const IGNORED_DEFAULT_INTEGRATIONS = [
"GlobalHandlers" && // We will use the react-native internal handlers
 

 
I'd change that slightly to say

```js
...defaultIntegrations.filter(i => !IGNORED_DEFAULT_INTEGRATIONS.includes(i))
```

and somewhere at the top

```js
const IGNORED_DEFAULT_INTEGRATIONS = [
  "GlobalHandlers", // We will use the react-native internal handlers
  "Breadcrumbs", // We add it later, just not patching fetch
  "TryCatch" // We don't need this
]
```

File: SentryReactNative.podspec (hunk 2 above)
Comment (HazAT) [reply SentryReactNative.podspec]: That's `sentry-cocoa`

File: android/src/main/java/io/sentry/RNSentryModule.java (hunk 3 above)
Comment (HazAT) [reply android/src/main/java/io/sentry/RNSentryModule.java]: Yeah, this is our test crash function

**Pull request merged**